src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/cas2/DomainEventService.kt

```diff
@@ -107,6 +109,7 @@ class DomainEventService(
         createdAt = OffsetDateTime.now(),
         data = objectMapper.writeValueAsString(domainEvent.data),
         service = "CAS2",
+        triggeredByUserId = userService.getUserForRequestOrNull()?.id,
```

Comment:
as mentioned in other comment, this will always return `null` for CAS2.
We'd need to implement something like the `getCas2User` in `ApplicationNoteService` 

https://github.com/ministryofjustice/hmpps-approved-premises-api/blob/78389046ca19c0ce72761a60cac89d81996076f9/src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/cas2/ApplicationNoteService.kt#L150 

I wonder if for now so that we aren't blocking you, we just set it to `null`? We could add it to our backlog, or I guess if we find we need it we would come around to fixing it naturally...

Reply:
That makes sense, i'll null it out for CAS2/3 for now to keep things simple 👍